File: client/packages/types/validation/ob-and-pregnancy-add.ts
import { z } from "zod";

export const obAndPregnancyAddSchemaa = z.object({
	patient_id: z.string(),
	age_onset_of_menses: z.number(),
	age_at_menopause: z.number(),
	comments_ob: z.string(),
	total_pregnancy: z.number(),
	full_term: z.number(),
	pre_term: z.number(),
	miscarriages: z.number(),
	living: z.number(),
	comments_pregnancy: z.string(),
});

File: server/patient/src/dto/history/obandpregnancy/obandpregnancy_add.rs
use common::dto::last_updated_input::LastUpdatedInput;
use serde::{Deserialize, Serialize};
use ts_rs::TS;
use validator::Validate;

#[derive(Deserialize, Serialize, Debug, Clone, Validate, Default, PartialEq, TS, Eq)]
#[ts(export, export_to = "../../bindings/OBandPregnancyAdd.ts")]
pub struct OBandPregnancyAdd {
    pub patient_id: String,
    pub age_onset_of_menses: u32,
    pub age_at_menopause: u32,
    pub comments_ob: Option<String>,
    pub total_pregnancy: Option<u32>,
    pub full_term: Option<u32>,
    pub pre_term: Option<u32>,
    pub miscarriages: Option<u32>,
    pub living: Option<u32>,
    pub comments_pregnancy: Option<String>,
    pub last_updated_input: LastUpdatedInput,
}
